library/psa_crypto_cipher.c

```diff
@@ -117,33 +116,34 @@ static psa_status_t mbedtls_cipher_validate_values(
 #if defined(PSA_WANT_ALG_CMAC)
         case PSA_ALG_CMAC:
 #endif
-#if defined(PSA_WANT_KEY_TYPE_AES) || defined(PSA_WANT_KEY_TYPE_ARIA) || \
-    defined(PSA_WANT_KEY_TYPE_DES) || defined(PSA_WANT_KEY_TYPE_CAMELLIA)
-    if (0
 #if defined(PSA_WANT_KEY_TYPE_AES)
-        || (key_type == PSA_KEY_TYPE_AES)
+    if (key_type == PSA_KEY_TYPE_AES) {
+        return PSA_SUCCESS;
+    }
 #endif
 #if defined(PSA_WANT_KEY_TYPE_ARIA)
-        || (key_type == PSA_KEY_TYPE_ARIA)
+    if (key_type == PSA_KEY_TYPE_ARIA) {
+        return PSA_SUCCESS;
+    }
 #endif
 #if defined(PSA_WANT_KEY_TYPE_DES)
-        || (key_type == PSA_KEY_TYPE_DES)
+    if (key_type == PSA_KEY_TYPE_DES) {
+        return PSA_SUCCESS;
+    }
 #endif
 #if defined(PSA_WANT_KEY_TYPE_CAMELLIA)
-        || (key_type == PSA_KEY_TYPE_CAMELLIA)
-#endif
-        ) {
-        break;
+    if (key_type == PSA_KEY_TYPE_CAMELLIA) {
+        return PSA_SUCCESS;
     }
-#endif /* PSA_WANT_KEY_TYPE_AES || ... */
+#endif
     return PSA_ERROR_NOT_SUPPORTED;
 #endif /* PSA_WANT_ALG_CTR || ... */
 
         default:
             return PSA_ERROR_NOT_SUPPORTED;
     }
 
-    return PSA_SUCCESS;
+    return PSA_ERROR_CORRUPTION_DETECTED;
```

Comment:
Actually is this line even reachable?

Reply:
No, it shouldn't be.

Reply:
Hmm, that can cause warnings from various compilers and also Coverity :(